### PR TITLE
[feature-wip](unique-key-merge-on-write) fix that IndexedColumnIterator next batch may return empty result

### PR DESCRIPTION
**Pull request by liaoxin01**
# Proposed changes

Issue Number: close #xxx

## Problem summary
F0819 11:08:01.494653 1043021 indexed_column_reader.cpp:249] Check failed: rows_to_read == rows_read  rows_to_read: 1 rows_read:0

When the key is between two pages，the result of `seek_at_or_after_value` will be the first row of next page. But `seek_at_or_after_value` doesn't move the page's offset, so next_batch will get an empty result.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by zhannngchen — DISMISSED** [review]
LGTM

File: be/src/olap/rowset/segment_v2/index_page.h [hunk 1]
```diff
@@ -138,6 +138,14 @@ class IndexPageIterator {
         return true;
     }
 
+    // Return true when has next page.
+    bool has_next() {
+        if ((_pos + 1) >= _reader->count()) {
```
Comment (cambyzju) [inline be/src/olap/rowset/segment_v2/index_page.h:143]: return (_pos + 1) < _reader->count();

Comment (liaoxin01) [reply be/src/olap/rowset/segment_v2/index_page.h]: fixed

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**